### ISSUE TITLE
Bandit `1.4.0` runtime error

### ISSUE THREAD

**Issue by timothyvanderaerden** [danschultzer/test_server#22]
When upgrading bandit to `1.4.0` I get the following error when running tests:

```
** (RuntimeError) Adapter functions must be called by stream owner
```

**Comment by danschultzer** [comment]
Hmm, yeah, there has been a lot of changes to the Bandit adapter. I'll push a PR soon, but got the WIP on this branch: https://github.com/danschultzer/test_server/tree/bandit-1-4

There are changes to how WebSocket is handled in Bandit too that I need to fix in that branch. I'll try get a fix out ASAP, but may be a few days as I'm underwater with work these days.

**Comment by danschultzer** [comment]
Also the reason we see this is that there is a difference between Bandit and other web servers handles the process. Bandit requires the caller to be the same pid as the plug. TestServer will run the callbacks from a GenServer. So I have to wrap all calls and push them to the plug process to make Bandit happy. It used to be only the HTTP2 adapter in Bandit, but now there's a uniform handling of it across the adapters.

**Comment by danschultzer** [comment]
Fixes in `v0.1.16`!
